### PR TITLE
[SPARK-41777][PYSPARK][ML] Integration testing for TorchDistributor

### PR DESCRIPTION
**Pull request by rithwik-db**
Just view the latest commit in this PR for the most accurate diff.

### What changes were proposed in this pull request?

Added integration tests for running distributed training on files.

### Why are the changes needed?

N/A

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

N/A

### CODE REVIEW

File: python/pyspark/ml/torch/tests/test_file.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# type: ignore
+
+batch_size = 100
```
Comment (WeichenXu123) [inline python/pyspark/ml/torch/tests/test_file.py:19]: shall we put the file in test resource directory ?

File: python/pyspark/ml/torch/tests/test_distributor.py [hunk 2]
```diff
@@ -286,6 +326,23 @@ def tearDown(self) -> None:
...
+            with self.subTest(f"subtest: {i + 1}"):
+                dist = TorchDistributor(num_processes, False, use_gpu)
+                dist._run_training_on_pytorch_file = lambda *args: os.environ.get(  # type: ignore
+                    CUDA_VISIBLE_DEVICES, "NONE"
+                )
+                self.assertEqual(
+                    expected,
+                    dist._run_distributed_training(dist._run_training_on_pytorch_file, "..."),
+                )
+
```
Comment (WeichenXu123) [inline python/pyspark/ml/torch/tests/test_distributor.py:345]: also add a test for running distributed training with train function input.

Comment (WeichenXu123) [reply python/pyspark/ml/torch/tests/test_distributor.py:345]: also add a test for local mode training.

Comment (rithwik-db) [reply python/pyspark/ml/torch/tests/test_distributor.py:345]: Added both in https://github.com/apache/spark/pull/39369

File: python/pyspark/ml/torch/tests/test_distributor.py [hunk 3]
```diff
@@ -288,6 +288,13 @@ def test_local_training_succeeds(self) -> None:
                 if cuda_env_var:
                     self.delete_env_vars({CUDA_VISIBLE_DEVICES: cuda_env_var})
 
+    def test_local_file_with_pytorch(self) -> None:
```
Comment (rithwik-db) [inline python/pyspark/ml/torch/tests/test_distributor.py:291]: As a sidenote, used `local-cluster` for testing `test_local_file_with_pytorch` and `test_distributed_file_with_pytorch`.

**Review by WeichenXu123 — APPROVED** [review]

**Review by lu-wang-dl — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: dev/requirements.txt [hunk 4]
```diff
@@ -59,3 +59,7 @@ googleapis-common-protos==1.56.4
 mypy-protobuf==3.3.0
 googleapis-common-protos-stubs==2.2.0
 grpc-stubs==1.24.11
+
+# TorchDistributor dependencies
+torch==1.13.1
+torchvision==0.14.1
```
Comment (HyukjinKwon) [inline dev/requirements.txt:65]: You actually don't need to pin this version. This is only for developement enviornment, and I would prefer to let other people test it with the latest versions.

Comment (dongjoon-hyun) [reply dev/requirements.txt:65]: +1 for @HyukjinKwon 's comment. And, this should be mentioned as test dependency explicitly.

File: python/pyspark/ml/torch/tests/test_distributor.py (hunk 3 above)
Comment (dongjoon-hyun) [reply python/pyspark/ml/torch/tests/test_distributor.py:291]: We should skip these tests when `pytorch` is not installed, @rithwik-db .

File: python/pyspark/ml/torch/tests/test_distributor.py [hunk 5]
```diff
@@ -349,6 +356,13 @@ def test_get_num_tasks_distributed(self) -> None:
 
         self.spark.sparkContext._conf.set("spark.task.resource.gpu.amount", "1")
 
+    def test_distributed_file_with_pytorch(self) -> None:
```
Comment (dongjoon-hyun) [inline python/pyspark/ml/torch/tests/test_distributor.py:359]: ditto.

**Pull request closed without merging**